### PR TITLE
[4.0] Fix multiple sampledata

### PR DESCRIPTION
**Pull request by Bakual**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/20662

### Summary of Changes
Adds a check to the sampledata JavaScript to ignore empty responses in the foreach loop.

I'm not sure what the current process is to generate the minimised JS files. Maybe @C-Lodder or @dgrammatiko can help here.

### Testing Instructions
* Install the testing sampledata plugin (or any other one) from https://github.com/joomla-extensions/testing-sample-data/releases/tag/v3.8.0. That one is expected to be broken in 4.0 but for this test it will do.
* Try installing the "Blog Sample Data" from the Control Panel.

### Expected result
Works

### Actual result
A JS Error prevents more than one step executed. See original issue for screenshots.

### Documentation Changes Required
None

### CODE REVIEW

**Comment by brianteeman** [comment]
Wrong branch??

**Comment by infograf768** [comment]
Yep should be towards 4.0 (staging is fine btw).
I did download the modified js file from the `Bakual:FixMultipleSampledata` branch and it now works fine.
Patch is adding
```

						if (value === null) {
							continue;
						}

```

**Comment by dgrammatiko** [comment]
> I'm not sure what the current process is to generate the minimised JS files. 

Just run `node build.js --compilejs` in the root folder (assuming npm install was already executed before, to setup the dependencies)

**Comment by Bakual** [comment]
> Wrong branch??

Aw crap sure. I saw it but was to tired to remember bevore hitting the create button.

**Comment by Quy** [comment]
> generate the minimised JS files

or use https://skalman.github.io/UglifyJS-online/

**Comment by Bakual** [comment]
Minifed file is updated as well now. Ready to test.

File: media/mod_sampledata/js/sampledata-process.js [hunk 1]
```diff
@@ -52,6 +52,11 @@ Joomla = window.Joomla || {};
 					// Display all messages that we got
 					for (var i = 0, l = response.data.length; i < l; i++) {
 						value   = response.data[i];
+
+						if (value === null) {
```
Comment (dgrammatiko) [inline media/mod_sampledata/js/sampledata-process.js:56]: you could also write this as `if (!value) {` which is exactly the same

**Comment by wilsonge** [comment]
LGTM! Thanks :)

**Pull request merged**